fv3core/utils/null_comm.py

```diff
@@ -59,3 +63,18 @@ def Recv(self, recvbuf, source, **kwargs):
 
     def Irecv(self, recvbuf, source, **kwargs):
         return NullAsyncResult(recvbuf)
+
+    def Split(self, color, key):
+        # key argument is ignored, assumes we're calling the ranks from least to
+        # greatest when mocking Split
+        self._split_comms[color] = self._split_comms.get(color, [])
+        rank = len(self._split_comms[color])
+        total_ranks = rank + 1
+        new_comm = NullComm(
+            rank=rank, total_ranks=total_ranks, fill_value=self._fill_value
+        )
+        for comm in self._split_comms[color]:
+            # won't know how many ranks there are until everything is split
+            comm.total_ranks = total_ranks
+        self._split_comms[color].append(new_comm)
```

Comment:
does this mean if we call this function twice with the same color we get two different communicators back? one of size 1 and one of size 2?

Reply:
Yes, this is normally meant to be used in a context where all the mocks are produced on one rank in mock-parallel. total_ranks gets updated on all comms of the same color when a new one is split, so if you call it twice, both will be of size 2.